src/libraries/System.Linq/tests/EmptyEnumerable.cs

```diff
@@ -39,5 +40,27 @@ public void EmptyEnumerableIsIndeedEmpty()
             TestEmptyEmpty<object>();
             TestEmptyEmpty<EmptyEnumerableTest>();
         }
+
+        [Fact]
+        public void ICollectionImplementationIsValid()
+        {
+            ICollection<int> c = Enumerable.Empty<int>() as ICollection<int>;
+            Assert.NotNull(c);
```

Comment:
Why not use `Assert.IsAssignableFrom<ICollection<int>>(...)` here?

Reply:
Changed